101-data-factory-hive-transformation/azuredeploy.json

```diff
@@ -0,0 +1,207 @@
+                "version": "3.2",
+                "timeToLive": "00:05:00",
+                "osType": "windows",
+                "linkedServiceName": "[variables('azureStorageLinkedServiceName')]"
+              }
+            }
+          },
+          {
+            "type": "datasets",
+            "name": "[variables('blobInputDatasetName')]",
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('azureStorageLinkedServiceName')]"
+            ],
+            "apiVersion": "2015-10-01",
+            "properties": {
+              "type": "AzureBlob",
+              "linkedServiceName": "[variables('azureStorageLinkedServiceName')]",
+              "typeProperties": {
+                "fileName": "[parameters('inputBlobName')]",
+                "folderPath": "[concat(parameters('blobContainer'), '/', parameters('inputBlobFolder'))]",
+                "format": {
+                  "type": "TextFormat",
+                  "columnDelimiter": ","
+                }
+              },
+              "availability": {
+                "frequency": "Month",
+                "interval": 1
+              },
+              "external": true
+            }
+          },
+          {
+            "type": "datasets",
+            "name": "[variables('blobOutputDatasetName')]",
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('azureStorageLinkedServiceName')]"
+            ],
+            "apiVersion": "2015-10-01",
+            "properties": {
+              "type": "AzureBlob",
+              "linkedServiceName": "[variables('azureStorageLinkedServiceName')]",
+              "typeProperties": {
+                "folderPath": "[concat(parameters('blobContainer'), '/', parameters('outputBlobFolder'))]",
+                "format": {
+                  "type": "TextFormat",
+                  "columnDelimiter": ","
+                }
+              },
+              "availability": {
+                "frequency": "Month",
+                "interval": 1
+              }
+            }
+          },
+          {
+            "type": "datapipelines",
+            "name": "[variables('pipelineName')]",
+            "dependsOn": [
+              "[variables('dataFactoryName')]",
+              "[variables('azureStorageLinkedServiceName')]",
+              "[variables('hdInsightOnDemandLinkedServiceName')]",
+              "[variables('blobInputDatasetName')]",
+              "[variables('blobOutputDatasetName')]"
+            ],
+            "apiVersion": "2015-10-01",
+            "properties": {
+              "description": "Pipeline that transforms data using Hive script.",
+              "activities": [
+                {
+                  "type": "HDInsightHive",
+                  "typeProperties": {
+                    "scriptPath": "[concat(parameters('blobContainer'), '/', parameters('hiveScriptFolder'), '/', parameters('hiveScriptFile'))]",
```

Comment:
The scriptPath gets resolved to adfgetstarted/script/partitionweblogs.hql. A prerequisite for this sample is to create **adfgetstarted** container, **script** folder, and upload the **partitionweblogs.hql** file to this folder.